Travis Build Failures: not enough credits

It has been recently identified that travis has been disabled for the last 1-2 months.

Admins are currently working to fully resolve the issue but travis ran in a PR and is failing. We need to look into the issues and fix the CI so it runs green again.

https://github.com/checkstyle/checkstyle/pull/10777
https://github.com/checkstyle/checkstyle/pull/10777/checks?check_run_id=3515099527

> 27971.4 | openjdk8 | DESC="NoErrorTest - Hibernate Search" | Linux | failed

````
[INFO] --- maven-enforcer-plugin:3.0.0-M3:enforce (enforce-common-rules) @ hibernate-search-parent ---
[WARNING] Rule 0: org.apache.maven.plugins.enforcer.RequireJavaVersion failed with message:
Detected JDK Version: 1.8.0-252 is not in the allowed range [11,).
[WARNING] Rule 1: org.apache.maven.plugins.enforcer.RequireMavenVersion failed with message:
Detected Maven Version: 3.6.3 is not in the allowed range 3.8.1.
[INFO] ------------------------------------------------------------------------
...
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M3:enforce (enforce-common-rules) on project hibernate-search-parent: Some Enforcer rules have failed. Look above for specific messages explaining why the rule failed. -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M3:enforce (enforce-common-rules) on project hibernate-search-parent: Some Enforcer rules have failed. Look above for specific messages explaining why the rule failed.
````

> 27971.5 | openjdk8 | DESC="NoErrorTest - checkstyle's sevntu" | Linux | failed

````
[INFO] --- maven-checkstyle-plugin:3.1.2:check (sevntu-checkstyle-check) @ checkstyle ---
[INFO] There are 2 errors reported by Checkstyle 9.1-SNAPSHOT with /home/travis/build/checkstyle/checkstyle/config/checkstyle_sevntu_checks.xml ruleset.
[ERROR] src/main/java/com/puppycrawl/tools/checkstyle/ThreadModeSettings.java:[29,33] (extension) AvoidDefaultSerializableInInnerClasses: Inner class should not implement default Serializable interface.
[ERROR] src/main/java/com/puppycrawl/tools/checkstyle/api/Violation.java:[49,5] (extension) AvoidDefaultSerializableInInnerClasses: Inner class should not implement default Serializable interface.
````

----------------------

Below is the correspondence with Travis Support to resolve the issue with travis not running:

Checkstyle:
````
We have just recently noticed that for the past 1-2 months our travis builds have not been running. When viewing https://app.travis-ci.com/github/checkstyle/checkstyle/requests it gives the following messages:

Owner checkstyle does not have enough credits.
Builds have been temporarily disabled for public repositories due to a negative credit balance. Please go to the Plan page to replenish your credit balance or alter your Consume paid credits for OSS setting. 

I don't understand why it says our plan changed on June 24, 2021, why we have credits, and why we are being charged credits now and not before?
````

Travis Support:
````
We have changed our plan few months back and notified all our customers about this new pricing plan changes[1]. Now we have free plan with 10K credits limit only. So for OSS credits you need to qualify with criteria. You can find information about our OSS  credits from this page[2]. 

We offer an Open Source Subscription for free to non-commercial open-source projects. 

Does this sound like you and your project? We'd be very happy to support you! 
````

Checkstyle:
````
I have looked over the OSS requirements and I do believe our project qualifies for it. I am an admin of the project.
````

Travis Support:
````
Open source has always been and always will be at the core of what Travis CI stands for. After reviewing your open source qualifications, I am pleased to notify you that 25k credits were added to your account allowing you to resume running OSS builds. When your credits begin running low again, please reach back out to the Support team.

Credits Consumption Metric
OS.              # CREDITS PER STARTED BUILD MINUTE
Linux.          10
Experimental FreeBSD.         10
Windows.         20
MacOS.         50


Please let us know if you have any questions! 

Thanks and happy building!,
````

https://docs.travis-ci.com/user/billing-overview/#partner-queue-solution
````
Partner Queue Solution is a solution for infrastructure sponsored by our Partners with OSS in mind which can be used completely for free. Currently, it includes:

    IBM CPU builds in IBM Cloud (sponsored by IBM)
    ARM64 CPU builds in Equinix Metal (former Packet) infrastructure (sponsored by ARM)

This is available only to Open Source Software repositories.

In order to run a job using Partner Queue Solution, use the following .travis.yml tags in your public repository:

os: linux
arch:
  - arm64
  - ppc64le
  - s390x

YAML

Please see our Build Environment overview and Building for Multiple CPU architectures pages for more details.

In order to start a build in the Usage based Plan, a positive credits balance is required in the account (at least 1 credit). The build job under Partner Queue Solution costs 0 credits per started minute. At the moment of introducing Partner Queue Solution active accounts on the Usage based Plans, including the Free Plan, with a balance of zero or fewer credits, balance is updated to hold 1 credit. Thus everybody can use Partner Queues without requesting Travis CI support to grant additional credits. If you run into a negative account balance after that, you still need to file an additional request.

In the case of Concurrency based Plans, you can use the above infrastructure for OSS builds without any credits in your account.
````

If any problem happening, there is hidden tab in Travis to show parse errors https://app.travis-ci.com/github/sevntu-checkstyle/checkstyle-samples/requests

--- a/.travis.yml
+++ b/.travis.yml
@@ -59,14 +59,13 @@ jobs:
         - CMD="$CMD1 && $CMD2"
         - USE_MAVEN_REPO="true"
 
-    # until https://github.com/checkstyle/checkstyle/issues/10778
-    # - jdk: openjdk8
-    #   env:
-    #     - DESC="NoErrorTest - Hibernate Search"
-    #     - CMD1="mvn -e --no-transfer-progress clean install -Pno-validations"
-    #     - CMD2="./.ci/wercker.sh no-error-hibernate-search"
-    #     - CMD="$CMD1 && $CMD2"
-    #     - USE_MAVEN_REPO="true"
+    - jdk: openjdk8
+      env:
+        - DESC="NoErrorTest - Hibernate Search"
+        - CMD1="mvn -e --no-transfer-progress clean install -Pno-validations"
+        - CMD2="./.ci/wercker.sh no-error-hibernate-search"
+        - CMD="$CMD1 && $CMD2"
+        - USE_MAVEN_REPO="true"
 
     # until https://github.com/sevntu-checkstyle/sevntu.checkstyle/pull/858
     # - jdk: openjdk8

--- a/.travis.yml
+++ b/.travis.yml
@@ -13,14 +13,13 @@ addons:
     packages:
       - xsltproc
       - xmlstarlet
-      - ruby
 
 branches:
   only:
     - master
 
 install:
-  - gem install chef-utils:16.6.14 mdl
+  -
 
 jobs:
   fast_finish: true

